File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/CycleStreetsRoutingTask.java
package net.cyclestreets.routing;

import net.cyclestreets.view.R;
import net.cyclestreets.content.RouteData;

import android.content.Context;

class CycleStreetsRoutingTask extends RoutingTask<Waypoints>
{
	/////////////////////////////////////////////////////
	private final String routeType_;
	private final int speed_;

	CycleStreetsRoutingTask(final String routeType,
				                  final int speed,
				                  final Context context)
	{
	  super(R.string.route_finding_new, context);
		routeType_ = routeType;
		speed_ = speed;
	} // NewRouteTask

	@Override
	protected RouteData doInBackground(final Waypoints... waypoints)
	{
	  final Waypoints wp = waypoints[0];
		return fetchRoute(routeType_, speed_, wp);
	} // doInBackgroud
} // NewRouteTask

File: libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/CycleStreetsRoutingTask.kt
package net.cyclestreets.routing

import android.content.Context

import net.cyclestreets.content.RouteData
import net.cyclestreets.view.R

internal open class CycleStreetsRoutingTask(private val routeType: String,
                                            private val speed: Int,
                                            context: Context,
                                            private val distance: Int? = null,
                                            private val duration: Int? = null,
                                            private val poiTypes: String? = null,
                                            private val saveRoute: Boolean = true) : RoutingTask<Waypoints>(R.string.route_finding_new, context) {
    override fun doInBackground(vararg waypoints: Waypoints): RouteData? {
        val wp = waypoints[0]
        return fetchRoute(routeType,
                          speed = speed,
                          waypoints = wp,
                          distance = distance,
                          duration = duration,
                          poiTypes = poiTypes,
                          saveRoute = saveRoute)
    }
}
